cmd/torpedo/torpedo.go

```diff
@@ -335,15 +338,32 @@ func (t *torpedo) validateContext(ctx *scheduler.Context) error {
 		)
 	}
 
+	GenerateSupportBundle(t, ctx)
 	if err := t.validateVolumes(ctx); err != nil {
+		GenerateSupportBundle(t, ctx)
 		return err
 	}
 	if err = t.s.WaitForRunning(ctx); err != nil {
+		GenerateSupportBundle(t, ctx)
 		return err
 	}
 	return err
 }
 
+// GenerateSupportBundle gathers logs and any artifacts pertinent to the scheduler and dumps them in the defined location
+func GenerateSupportBundle(t *torpedo, ctx *scheduler.Context) error {
```

Comment:
why not make this a method on torpedo instance? also it could be a private method, right?

Reply:
What are we trying to prevent accessing this? IMO In fact this should be available eventually if we expose APIs outside too.